### ISSUE TITLE
1.2: Error single quotes in body tag attribute after upgrade

### ISSUE THREAD

**Issue by gjr-qm** [meteor/meteor#5232]
When using single quotes in an attribute value of the main html file, i get an error (Uncaught SyntaxError: missing ) after argument list) in the compiled version. This error did not occur in version 1.2.0.0.

Extra info: the problem only occurs on the body tag.

For example:
`<body onclick="alert('hello'); "> // -> error`

Workaround for now is escaping the quotes:
`<body onclick="alert(\'hello\'); "> // -> no error`

**Comment by glasser** [comment]
I was unable to reproduce this.

Please provide a full reproduction as described in: https://github.com/meteor/meteor/wiki/Contributing-to-Meteor#reporting-a-bug-in-meteor  Without a way to replicate your problem, we can't debug and fix it.

Perhaps this is static-html vs blaze-html-templates?

**Comment by gjr-qm** [comment]
Thanks for your reply. This bug for me is not big enough to make a full reproduction as the rules subscribe.

To reproduce the bug:
- start new meteor app: meteor create testapp
- cd testapp
- edit testapp.html, add `onclick="alert('hello');"` to the body tag
- meteor testapp

Meteor bootstrap will fail, because of single quotes in body tag attributes.

**Comment by glasser** [comment]
I did try that (again) and it did work for me. Please do follow the suggestions at that link.

**Comment by gjr-qm** [comment]
Ok, so i created a repo and here's how to reproduce the bug (input bold):

**git clone https://github.com/gjr-qm/meteor-single-quote-bug.git**
Cloning into 'meteor-single-quote-bug'...
remote: Counting objects: 15, done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 15 (delta 1), reused 11 (delta 0), pack-reused 0
Unpacking objects: 100% (15/15), done.
Checking connectivity... done.
**cd meteor-single-quote-bug**
**meteor**

App starts running and when you open your browser it will show the error:
Uncaught SyntaxError: missing ) after argument list

I think  the bug is caused by the code in caching-html-compiler.js, line 111, where the value of the key-value pairs is not escaped, but just added to the string.

**Comment by hynese** [comment]
+1

[Meteor 1.2.0.1, OSX Yosemite]

As a temporary measure, I would suggest turning the single and double quotes inside-out:

```
<body ng-app='simple-todos'
      ng-include='"todos-list.ng.html"'
      ng-controller='TodosListCtrl'>
</body>
```

**Comment by glasser** [comment]
OK, thanks for explaining that the issue was *in the browser*; my interpretation of the original bug report was that this was a build time failure which I did not observe.

**Comment by stubailo** [comment]
I've pushed a fix, hopefully it will go out in the next patch release.

**Comment by aaronroberson** [comment]
To get around this I added the template path to the controller and then accessed the path from the view model instead of using a static string. 

JS
```
some.controller.js
class SomeController {
  constructor() {
  this.templateUrl = 'client/some.ng.html';
  }
}
```
HTML
```
<ng-include="vm.templateUrl">
```

**Comment by stubailo** [comment]
Fixed in newly released Meteor 1.2.0.2! Let me know if it isn't :]
